nrf52-hal-common/src/twim.rs

```diff
@@ -110,151 +112,151 @@ impl<T> Twim<T> where T: TwimExt {
 
     /// Write to an I2C slave
     ///
-    /// The buffer must have a length of at most 255 bytes on the nRF52832
-    /// and at most 65535 bytes on the nRF52840.
     pub fn write(&mut self,
         address: u8,
         buffer:  &[u8],
     )
         -> Result<(), Error>
     {
-
-        if buffer.len() > EASY_DMA_SIZE {
-            return Err(Error::TxBufferTooLong);
+        if SRAM_LOWER <= ( buffer.as_ptr() as usize ) && (buffer.as_ptr() as usize) < SRAM_UPPER {
```

Comment:
Instead of using the address of the very first element, this could check if the end of the buffer is still at an address <SRAM_UPPER.

Reply:
Still, good catch. just adding the `buffer.len()` should work.

Reply:
If this is an issue there are other problems with the buffer halfway outside of the RAM, but I makes sense include buffer length for readability.